message_parser/src/main/java/cumulus_message_adapter/message_parser/MessageAdapter.java

```diff
@@ -0,0 +1,127 @@
+package cumulus_message_adapter.message_parser;
+
+import com.amazonaws.services.lambda.runtime.Context; 
+
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.util.Scanner;
+import java.util.Map;
+import java.util.HashMap;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+public class MessageAdapter implements IMessageAdapter
+{
+    /**
+     * Call to message adapter zip to execute a message adapter function. Pass args through the process input
+     * and read return result from process output.
+     * @param messageAdapterFunction - 'loadRemoteEvent', 'loadNestedEvent', or 'createNextEvent'
+     * @param inputJson - argument to message adapter function. Json that contains all of the params.
+     * @return the return from the message adapter function
+     */
+    public String CallMessageAdapterFunction(String messageAdapterFunction, String inputJson)
+        throws MessageAdapterException
+    {
+        String messageAdapterOutput = "";
+
+        try
+        {
+            ProcessBuilder processBuilder = new ProcessBuilder("python", "cumulus-message-adapter.zip", messageAdapterFunction);
+
+            Process process = processBuilder.start();
+
+            OutputStreamWriter writer = new OutputStreamWriter(process.getOutputStream());
+            writer.write(inputJson);
+            writer.close();
+
+            // Log the entire error
+            // TO DO: Update logging
+            Scanner scanner = new Scanner(process.getErrorStream());
+            Boolean hasError = false;
+            while(scanner.hasNextLine()) 
+            {
+                hasError = true;
+                System.out.println(String.format("Cumulus Message Adapter error: %s: %s", messageAdapterFunction, scanner.nextLine()));  
+            }
+            scanner.close();
+
+            if(hasError)
+            {
+                throw new MessageAdapterException("Error executing " + messageAdapterFunction);
+            }
+
+            scanner = new Scanner(process.getInputStream());
+            if(scanner.hasNextLine()) 
+            {
+                messageAdapterOutput = scanner.nextLine();
+            }
+            scanner.close();
+        }
+        catch(IOException e)
+        {
+            throw new MessageAdapterException("Unable to find Cumulus Message Adapter", e.getCause());      
+        }
+
+        return messageAdapterOutput;        
+    }
+
+    /**
+     * Format the arguments and call the 'loadRemoteEvent' message adapter function
+     * @param eventJson - Json passed from lambda
+     * @return result of 'loadRemoteEvent'
+     */
+    public String LoadRemoteEvent(String eventJson)
```

Comment:
The JS message-adapter accepts Objects as the event and handles the JSON conversion.  The fact that the messages are passed back and forth between the adapter and the python script as JSON is an implementation detail that we don't need to expose to users of this script.  By using JSON for this argument, we're forcing users to have to convert the input to JSON themselves.

Reply:
The same goes for the return value.

Reply:
Yeah for Java it is String and String. Talked with Alireza about this weeks ago and that was what podaac was using.